carbon-design-system/carbon-addons-iot-react
# feat(table): add prop to allow search to always be open

Closes #3186 

**Summary**

- adds `isExpanded` prop to Table search props to allow the search field to always be open

**Change List (commits, features, bugs, etc)**

- updated prop types
- added isExpanded prop
- added isExpanded knob to table playground story
- added tests to confirm
- updated snapshots

**Acceptance Test (how to verify the PR)**

- go to the playground story
- turn on the `isExpanded` knob on the Search tab
- confirm the search input stays open regardless of focus or content
- turn off the `isExpanded` knob. Confirm search input opens on focus, stays open with content, and closes when there is no content and no focus

**Regression Test (how to make sure this PR doesn't break old functionality)**

- confirm search continues to function as expected



**Things to look for during review**
